daemon/api_prompting.go

```diff
@@ -97,6 +100,186 @@ func getUserID(r *http.Request) (uint32, Response) {
 	return uint32(userIDInt), nil
 }
 
+type invalidReason string
+
+const (
+	unsupportedValueReason invalidReason = "unsupported-value"
+	parseErrorReason       invalidReason = "parse-error"
+)
+
+type invalidFieldValue struct {
+	Reason   invalidReason `json:"invalid-reason"`
+	Metadata interface{}   `json:"metadata,omitempty"`
+}
+
+type UnsupportedValueValue struct {
+	err *prompting_errors.UnsupportedValueError
+}
+
+func (v *UnsupportedValueValue) MarshalJSON() ([]byte, error) {
+	value := make(map[string]invalidFieldValue, 1)
+	value[v.err.Field] = invalidFieldValue{
+		Reason: unsupportedValueReason,
+		Metadata: &struct {
+			Received  interface{} `json:"received-invalid"`
+			Supported []string    `json:"supported"`
+		}{
+			Received:  v.err.Unsupported,
+			Supported: v.err.Supported,
+		},
+	}
+	return json.Marshal(value)
+}
+
+type ParseErrorValue struct {
+	err *prompting_errors.ParseError
+}
+
+func (v *ParseErrorValue) MarshalJSON() ([]byte, error) {
+	value := make(map[string]invalidFieldValue, 1)
+	value[v.err.Field] = invalidFieldValue{
+		Reason: parseErrorReason,
+		Metadata: &struct {
+			Received string `json:"received-invalid"`
+			// TODO: once documentation exists for user-defined fields
+			// DocumentationURL string `json:"documentation"`
+		}{
+			Received: v.err.Invalid,
+			// TODO: once documentation exists for user-defined fields
+			// DocumentationURL: <url>,
+		},
+	}
+	return json.Marshal(value)
+}
+
+type RequestedPathNotMatchedErrorJSON prompting_errors.RequestedPathNotMatchedError
```

Comment:
does not need to be exported
```suggestion
type promptingRequestedPathNotMatchedError prompting_errors.RequestedPathNotMatchedError
```